### DISCUSSION TITLE
Does Vercel offer a vector database to use with OpenAi?

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#55030] [Help]
### Summary

I am interesting on building a chatbot using Next.

Few questions:

Does Vercel offers a vector database to integrate with OpenAi? If not, would you mention some popular ones?

Would you have some code samples for OpenAi UI?

What exactly are the Vercel SDK's?

**Comment by devjiwonchoi** [comment]
There's a template with Vercel PSQL and pgvector you can check out if this is what you want.

https://vercel.com/templates/next.js/postgres-pgvector

**Reply by devjiwonchoi** [reply]
Also check out [Vercel AI SDK](https://github.com/vercel/ai) and [example chatbot](https://github.com/vercel-labs/ai-chatbot)

**Reply by Elindo586** [reply]
Thank you much....  I saw that...

Now this is coming like a truck to me and first time ever reading those words...

I assume a vector database is not these items below... but the items below would work like a vector database to connect the AI?

And what exactly these things do?

Postgres + Prisma + pgvector ???

**Reply by devjiwonchoi** [reply]
Postgres: RDBMS (Relational Database like MySQL)
Prisma: ORM (Object Relational Mapping) - Converts DB schema to language understandable, provides declarative & JSON to manage SQL queries like `SELECT ...`.
pgvector: Open-source vector similarity search for Postgres

**Reply by devjiwonchoi** [reply]
Vercel AI SDK provides handy features to handle multiple AI APIs such as OpenAI, which lets you easily implement on front-end frameworks like Next.js.

See https://sdk.vercel.ai/docs for more information.
